server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/TransformTreeStage.java

```diff
@@ -141,7 +143,7 @@ protected List<Node> transformAST(AnalysisContext ctx,
   private void shapeSectionsAndParagraphs(Node parent) {
     LinkedList<Node> stack = new LinkedList<>();
     for (Node node : parent.getChildren()) {
-      parent.removeChild(node);
+      parent.getChildren().remove(0);
```

Comment:
Why should this be equivalent?

Reply:
ok, I think I understand why it works. That being said, how many copies are made by the CopyOnWriteArrayList while the children are being added and removed? would it be possible to to accumulate the changes and exchange the content at once?